### PR TITLE
[issue-328] jsonyamlxml: fix parsing of file copyright text

### LINKED ISSUE #328: json parsing issue when no packages are present
Parsing a json document with no or empty `packages` results in the following error during file parsing:
```
FILE_COPYRIGHT_TEXT Can not appear before FILE_NAME
```

### PR DESCRIPTION
**Pull request by meretp**
fixes #328

Signed-off-by: Meret Behrens <meret.behrens@tngtech.com>

### CODE REVIEW

File: spdx/parsers/jsonyamlxmlbuilders.py [hunk 1]
```diff
@@ -205,6 +205,22 @@ def set_file_type(self, doc, type_value):
 
         return super(FileBuilder, self).set_file_type(doc, f"namespace#fileType_{type_value.lower()}")
 
+    def set_file_copyright(self, doc, text):
+        """
+        Raise OrderError if no package or file defined.
```
Comment (armintaenzertng) [inline spdx/parsers/jsonyamlxmlbuilders.py:210]: Luckily, packages are out of the equation now :)

File: spdx/parsers/jsonyamlxmlbuilders.py [hunk 2]
```diff
@@ -205,6 +205,22 @@ def set_file_type(self, doc, type_value):
...
+        """
+        if self.has_file(doc):
+            if not self.file_copytext_set:
+                self.file_copytext_set = True
+                self.file(doc).copyright = text
+                return True
+            else:
+                raise CardinalityError("File::CopyRight")
+        else:
+            raise OrderError("File::CopyRight")
```
Comment (armintaenzertng) [inline spdx/parsers/jsonyamlxmlbuilders.py:221]: early return could be applied

Comment (meretp) [reply spdx/parsers/jsonyamlxmlbuilders.py]: applied

File: spdx/parsers/jsonyamlxmlbuilders.py (hunk 1 above)
Comment (meretp) [reply spdx/parsers/jsonyamlxmlbuilders.py]: deleted

**Review by armintaenzertng — DISMISSED** [review]

**Review by armintaenzertng — APPROVED** [review]

**Pull request merged**